rest/blip_api_test.go

```diff
@@ -1449,54 +1450,70 @@ func TestPutAttachmentViaBlipGetViaRest(t *testing.T) {
 }
 
 func TestPutAttachmentViaBlipGetViaBlip(t *testing.T) {
-
 	defer base.SetUpTestLogging(base.LevelInfo, base.KeyHTTP, base.KeySync, base.KeySyncMsg)()
 
-	// Create blip tester
-	bt, err := NewBlipTesterFromSpec(t, BlipTesterSpec{
-		connectingUsername:          "user1",
-		connectingPassword:          "1234",
-		connectingUserChannelGrants: []string{"*"}, // All channels
-	})
-	require.NoError(t, err, "Unexpected error creating BlipTester")
-	defer bt.Close()
-
-	attachmentBody := "attach"
-	digest := db.Sha1DigestKey([]byte(attachmentBody))
+	var tests = []struct {
+		name                   string
+		inputBlipProtocol      string
+		inputAttachmentVersion int64
+	}{
+		{"TestPutAttViaBlipGetViaBlipCBMobile2AttVer1", db.BlipCBMobileReplicationV2, db.LegacyAttachmentVersion},
+		{"TestPutAttViaBlipGetViaBlipCBMobile2AttVer2", db.BlipCBMobileReplicationV2, db.NonLegacyAttachmentVersion},
+		{"TestPutAttViaBlipGetViaBlipCBMobile3AttVer1", db.BlipCBMobileReplicationV3, db.LegacyAttachmentVersion},
+		{"TestPutAttViaBlipGetViaBlipCBMobile3AttVer2", db.BlipCBMobileReplicationV3, db.NonLegacyAttachmentVersion},
```

Comment:
To properly test these cases I expect you'd need multiple docs referencing the same attachment digest - otherwise you're not testing any of the new allowedAttachment indexing.

Reply:
Makes sense, the TestPutAttachmentViaBlipGetViaBlip has been refactored with two documents referencing the same attachment.